extensions/PhabBugz/lib/Revision.pm

```diff
@@ -0,0 +1,280 @@
+#               "status": "added",
+#               "isBlocking": false,
+#               "actorPHID": null
+#             },
+#             {
+#               "reviewerPHID": "PHID-USER-o5dnet6dp4dkxkg5b3ox",
+#               "status": "rejected",
+#               "isBlocking": false,
+#               "actorPHID": "PHID-USER-o5dnet6dp4dkxkg5b3ox"
+#             }
+#           ]
+#         },
+#         "subscribers": {
+#           "subscriberPHIDs": [],
+#           "subscriberCount": 0,
+#           "viewerIsSubscribed": true
+#         },
+#         "projects": {
+#           "projectPHIDs": []
+#         }
+#       }
+#     }
+#   ],
+#   "maps": {},
+#   "query": {
+#     "queryKey": null
+#   },
+#   "cursor": {
+#     "limit": 100,
+#     "after": null,
+#     "before": null,
+#     "order": null
+#   }
+# }
+
+#########################
+#     Modification      #
+#########################
+
+sub update {
+    my ($self) = @_;
+
+    my $data = {
+        objectIdentifier => $self->phid,
+        transactions     => []
+    };
+
+    if ($self->{added_comments}) {
+        foreach my $comment (@{ $self->{added_comments} }) {
+            push(@{ $data->{transactions} }, {
+                type  => 'comment',
+                value => $comment
+            });
+        }
+    }
+
+    if ($self->{set_subscribers}) {
+        push(@{ $data->{transactions} }, {
+            type  => 'subscribers.set',
+            value => $self->{set_subscribers}
+        });
+    }
+
+    if ($self->{set_policy}) {
+        foreach my $name ("view", "edit") {
+            next unless $self->{set_policy}->{$name};
+            push(@{ $data->{transactions} }, {
+                type  => $name,
+                value => $self->{set_policy}->{$name}
+            });
+        }
+    }
+
+    request('differential.revision.edit', $data);
+}
+
+#########################
+#      Accessors        #
+#########################
+
+sub id              { $_[0]->{id};                          }
+sub phid            { $_[0]->{phid};                        }
+sub title           { $_[0]->{fields}->{title};             }
+sub status          { $_[0]->{fields}->{status}->{value};   }
+sub creation_ts     { $_[0]->{fields}->{dateCreated};       }
+sub modification_ts { $_[0]->{fields}->{dateModified};      }
+sub author_phid     { $_[0]->{fields}->{authorPHID};        }
+sub bug_id          { $_[0]->{fields}->{'bugzilla.bug-id'}; }
+
+sub view_policy { $_[0]->{fields}->{policy}->{view}; }
+sub edit_policy { $_[0]->{fields}->{policy}->{edit}; }
+
+sub reviewers_raw    { $_[0]->{attachments}->{reviewers}->{reviewers};          }
+sub subscribers_raw  { $_[0]->{attachments}->{subscribers};                    }
+sub projects_raw     { $_[0]->{attachments}->{projects};                       }
+sub subscriber_count { $_[0]->{attachments}->{subscribers}->{subscriberCount}; }
+
+sub bug {
+    my ($self) = @_;
+    my $bug = $self->{bug} ||= new Bugzilla::Bug($self->bug_id);
```

Comment:
This will always be called, since there would be no strong references to the bug. Remove the call to weaken().

Reply:
Also, don't use indirect object syntax in new code.